### PR TITLE
use release version of `agent-payload`

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

https://github.com/DataDog/datadog-agent/pull/10650 introduced a dependency on payload v4.0.89 which is most likely an error. This PR fixes this issue, and also makes us using a released version of the agent payload and not a commit based version.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -42,7 +42,7 @@ require (
 	code.cloudfoundry.org/garden v0.0.0-20210208153517-580cadd489d2
 	code.cloudfoundry.org/lager v2.0.0+incompatible
 	github.com/CycloneDX/cyclonedx-go v0.7.1
-	github.com/DataDog/agent-payload/v5 v5.0.90-0.20230717070528-bfb5d051f2ab
```
Comment (truthbk) [inline go.mod:45]: So this was introduced here: https://github.com/DataDog/datadog-agent/pull/18165/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R45, you might want to talk to @Gui774ume

File: pkg/logs/internal/processor/proto.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ package processor
 import (
 	"time"
 
-	"github.com/DataDog/agent-payload/pb"
+	"github.com/DataDog/agent-payload/v5/pb"
```
Comment (truthbk) [inline pkg/logs/internal/processor/proto.go:11]: Yes, this makes sense, indeed my bad.

**Review by truthbk — COMMENTED** [review]
Makes sense, specifically the changes to `pkg/logs/internal/processor`; I wish we'd have had tests where this could've been caught.

File: go.mod (hunk 1 above)
Comment (paulcacheux) [reply go.mod:45]: I'm in his team, so it's good to go like this. This commit is included in the new release

**Review by truthbk — APPROVED** [review]

**Review by scottopell — APPROVED** [review]

**Pull request merged**